src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1292,6 +1281,42 @@ CompressDictMatMulf32::CompressDictMatMulf32(Context::Ref ctx) {
     register_matcher(std::make_shared<opp::Matcher>(res, "OptCompressDictMatMulf32"), std::move(callback));
 }
 
+//
+// FROM:
+//     Param [1x4x1023x64] -> ShapeOf -> ...
+//
+// TO:
+//     Const [4] -> ...
+//
+ShapeOfToConst::ShapeOfToConst(const std::shared_ptr<ov::Model>& model) {
+    auto key_val = opp::wrap_type<ov::op::v0::Parameter>();
+    auto shapeof = opp::wrap_type<ov::op::v3::ShapeOf>({key_val});
```

Comment:
This pattern is not generic enough again.. What you need to look for is `wrap_type<ShapeOf>({any_input()})`.

That's quite generic and can get rid of such problems in the future, but is quite invasive too so it is interesting how it impacts our current execution.

Reply:
What if we face `...`->`Non-input node`->`ShapeOf`->`...`?

Reply:
I don't understand why it needs to be generic? It serves a particular purpose for which it works just fine